### PR TITLE
For #18507: Prevent screenshots on credit card screens.

### LINKED ISSUE #18507: Ensure the Credit card screens cannot be screenshot
Note that there is an existing PR with feedback and take aways https://github.com/mozilla-mobile/fenix/pull/18890. Please consider using it.

-----

Secure secure flags so that the windows of the Credit cards screen cannot be screenshotted.

### PR DESCRIPTION
**Pull request by mcarare**
Fixes #18507 

### Pull Request checklist

- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/shared-docs/blob/master/android/accessibility_guide.md) or does not include any user facing features. In addition, it includes a screenshot of a successful [accessibility scan](https://play.google.com/store/apps/details?id=com.google.android.apps.accessibility.auditor&hl=en_US) to ensure no new defects are added to the product.

### To download an APK when reviewing a PR:
1. click on Show All Checks,
2. click Details next to "Taskcluster (pull_request)" after it appears and then finishes with a green checkmark,
3. click on the "Fenix - assemble" task, then click "Run Artifacts".
4. the APK links should be on the left side of the screen, named for each CPU architecture

### CODE REVIEW

**Review by codrut-topliceanu — APPROVED** [review]
👍 Works + Looks good to me.

**Comment by gabrielluong** [comment]
Can we file an issue about using the new `SecureFragment` in the logins? Thanks!

File: app/src/main/java/org/mozilla/fenix/ext/Fragment.kt [hunk 1]
```diff
@@ -90,3 +91,15 @@ fun Fragment.breadcrumb(
...
+    this.activity?.window?.addFlags(
+        WindowManager.LayoutParams.FLAG_SECURE
+    )
+}
+
+fun Fragment.removeSecure() {
+    this.activity?.window?.clearFlags(
+        WindowManager.LayoutParams.FLAG_SECURE
+    )
+}
```
Comment (gabrielluong) [inline app/src/main/java/org/mozilla/fenix/ext/Fragment.kt:105]: Can you add kDocs for these 2 methods?

**Review by gabrielluong — APPROVED** [review]

**Comment by mcarare** [comment]
> Can we file an issue about using the new `SecureFragment` in the logins? Thanks!

Filed https://github.com/mozilla-mobile/fenix/issues/19567.

**Pull request merged**